BezierKit/Library/Path+Data.swift

```diff
@@ -24,14 +22,30 @@ fileprivate extension Data {
     }
 }
 
-fileprivate extension InputStream {
-    func readNativeValue<T>(_ value: UnsafeMutablePointer<T>) -> Bool {
+fileprivate struct DataStream {
+    var dataCursor: Data.Index
+    let data: Data
```

Comment:
would it work just to have a pointer to the data's bytes here instead of a reference to the data and a cursor? the pointer could be advanced to move the cursor and the data could be copied from the pointer?